cmds/mknod/mknod_test.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2016-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+	"testing"
+)
+
+var (
+	testPath = "."
+	// if true removeAll the testPath on the end
+	remove = true
+)
+
+type test struct {
+	args    []string
+	expects string
+}
+
+func run(c *exec.Cmd) (string, string, error) {
+	var o, e bytes.Buffer
+	c.Stdout, c.Stderr = &o, &e
+	err := c.Run()
+	return o.String(), e.String(), err
+}
+
+func TestMknodFifo(t *testing.T) {
+	tempDir, err := ioutil.TempDir("", "TestMknodFIFO")
```

Comment:
You can use the `github.com/u-root/u-root/shared/testutil` package to create a temp directory and build the binary.